apps/explorer/src/hooks/useImage.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { useCallback, useLayoutEffect, useRef, useState } from 'react';
+
+import { useImageMod } from './useImageMod';
+
+type Status = 'loading' | 'failed' | 'loaded';
+
+interface UseImageProps {
+    src?: string;
+    moderate?: boolean;
+}
+
+export function useImage({ src = '', moderate = true }: UseImageProps) {
+    const [status, setStatus] = useState<Status>('loading');
+    const formatted = src?.replace(/^ipfs:\/\//, 'https://ipfs.io/ipfs/');
+    const { data: allowed, isFetched } = useImageMod({
+        url: formatted,
+        enabled: moderate,
+    });
+
+    const ref = useRef<HTMLImageElement | null>(null);
+
+    const cleanup = () => {
+        if (ref.current) {
+            ref.current.onload = null;
+            ref.current.onerror = null;
+            ref.current = null;
+        }
+    };
+
+    const load = useCallback(() => {
```

Comment:
Should this set the status to loading for cases where the `src` changes and it loads a new image?